.gitignore

```diff
@@ -1,23 +1,23 @@
-/.idea
+# Ignore everything
+/*
 
-/work
-/tools
-/build
-/.gradle
-*.log
-*.reject
-/run/
-
-*.iml
-*.ipr
-*.iws
-*.ini
-/out
-/classes
-/test
-
-/bin/
-/.classpath
-/.project
-/.settings
+# Folders
+!/gradle
+!/src
+!/.github
 
+# File
+!/.gitignore
+!/.gitmodules
+!/build.gradle
+!/gradle.properties
+!/settings.gradle
+!/gradlew
+!/gradlew.bat
+!/HEADER
+!/Jenkinsfile
+!/l4j-manifest.xml
+!/LICENSE
+!/README.md
+!/wrapper.nim
```

Comment:
I would put a comment in the gitignore signifying native source files vs java source files.